controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go

```diff
@@ -147,6 +148,7 @@ func (in *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
 		{spec, kubeadmConfigSpec, clusterConfiguration, controllerManager, "*"},
 		{spec, kubeadmConfigSpec, clusterConfiguration, scheduler, "*"},
 		{spec, kubeadmConfigSpec, initConfiguration, nodeRegistration, "*"},
+		{spec, kubeadmConfigSpec, initConfiguration, skipPhases, "*"},
```

Comment:
Would the same make sense on the joinConfiguration?

Let's please also adjust our unit test (there are already a few cases for various fields, let's add this one)

Reply:
xref: https://kubernetes.slack.com/archives/C8TSNPY4T/p1669640280683049?thread_ts=1669638055.159729&cid=C8TSNPY4T

Reply:
Even though initConfiguration is only used for the first Machine I think it should be okay to have it mutable in cases where a user would want to modify the field before the first Machine comes up

Reply:
yeap makes sense @sbueringer  ^^ pushed changed